packages/legend-studio/src/components/editor/edit-panel/service-editor/ServiceRegistrationEditor.tsx

```diff
@@ -0,0 +1,232 @@
+import { ServiceExecutionMode } from '@finos/legend-graph';
+import { flowResult } from 'mobx';
+import { Version } from '@finos/legend-server-sdlc';
+import { useEditorStore } from '../../EditorStoreProvider';
+import { useApplicationStore } from '@finos/legend-application';
+
+export const ServiceRegistrationEditor = observer(() => {
+  const editorStore = useEditorStore();
+  const applicationStore = useApplicationStore();
+  const serviceState = editorStore.getCurrentEditorState(ServiceEditorState);
+  const registrationState = serviceState.registrationState;
+  // service servers
+  const envOptions = registrationState.options.map((info) => ({
+    label: info.env.toUpperCase(),
+    value: info.env,
+  }));
+  const selectedEnvOption = registrationState.serviceEnv
+    ? {
+        label: registrationState.serviceEnv.toUpperCase(),
+        value: registrationState.serviceEnv,
+      }
+    : null;
+  // service types
+  const serviceTypesOptions = registrationState.executionModes.map((mode) => ({
+    label: prettyCONSTName(mode),
+    value: mode,
+  }));
+  const selectedServiceType = registrationState.serviceExecutionMode
+    ? {
+        label: prettyCONSTName(registrationState.serviceExecutionMode),
+        value: registrationState.serviceExecutionMode,
+      }
+    : null;
+  const onServiceTypeSelectionChange = (
+    val: { label: ServiceExecutionMode; value: ServiceExecutionMode } | null,
+  ): void => {
+    registrationState.updateType(val?.value);
+  };
+  const onServerEnvChange = (
+    val: { label: string; value: string } | null,
+  ): void => {
+    registrationState.updateEnv(val?.value);
+  };
+  // version Projection
+  const selectedVersion = registrationState.projectVersion
+    ? {
+        label:
+          registrationState.projectVersion instanceof Version
+            ? registrationState.projectVersion.id.id
+            : registrationState.projectVersion,
+        value: registrationState.projectVersion,
+      }
+    : null;
+  const onVersionSelectionChange = (
+    val: { label: string; value: Version | string } | null,
+  ): void => {
+    registrationState.setProjectVersion(val?.value);
+  };
+  const versionPlaceholder =
+    registrationState.versionOptions === undefined
+      ? `Only valid for ${prettyCONSTName(
+          ServiceExecutionMode.SEMI_INTERACTIVE,
+        )} and ${prettyCONSTName(ServiceExecutionMode.PROD)} service types`
+      : !registrationState.versionOptions.length
+      ? 'Project has no versions'
+      : undefined;
+  // activate
+  const toggleActivatePostRegistration = (): void => {
+    registrationState.setActivatePostRegistration(
+      !registrationState.activatePostRegistration,
+    );
+  };
+  // actions
+  const registerService = (): void => {
+    if (selectedEnvOption && selectedServiceType) {
+      flowResult(registrationState.registerService()).catch(
+        applicationStore.alertUnhandledError,
+      );
+    }
+  };
+  const disableRegistration =
```

Comment:
Why don't I see in the demo video the buton being disabled when you register the service?

Reply:
Video was for demo purposes. this code was commented out.